dot/sync/message.go

```diff
@@ -338,8 +338,14 @@ func (s *Service) handleChainByHash(ancestor, descendant common.Hash,
 		return nil, err
 	}
 
+	// If the direction is descending, prune from the start.
+	// if the direction is ascending it should prune from the end.
 	if uint(len(subchain)) > max {
-		subchain = subchain[:max]
+		if direction == network.Ascending {
+			subchain = subchain[:max]
+		} else {
+			subchain = subchain[uint(len(subchain))-max:]
```

Comment:
nit: `len` will be always greater than or equal 0
```suggestion
			subchain = subchain[len(subchain)-max:]
```

Reply:
It's actually interesting the index can be an `uint`, I thought these would need to be `int`.